alpa/device_mesh.py

```diff
@@ -274,6 +276,10 @@ def get_exec_total_allocation_size(self, uuid: int):
     def get_exec_grad_sync_channel_ids(self, uuid: int):
         return self.executables[uuid].grad_sync_channel_ids
 
+    def set_runtime_random_seed(self, seed: int):
+        for d in self.local_devices:
+            d.set_seed(seed)
```

Comment:
If we set the same seed for all devices, will we generate all the same random numbers (e.g., same weights & dropout masks) on the same device? Same question for seeds on different hosts and different device meshes below.

Reply:
The current design:
1. We set the same "initial" random seed for all devices in all meshes.
2. When a device generates random states. It uses `actual_seed = initial ^ (device_id_in_a_mesh + mesh_id * 1024)`. Currently I only implemented `initial ^ device_id_in_a_mesh` part (see https://github.com/alpa-projects/tensorflow-alpa/blob/6741748e6f214dc8000a4bd24967df642852bb1a/tensorflow/compiler/xla/service/gpu/rng_thunk.cc#L27). I will add the `mesh_id` support when I implement this for `PipeshardParallel`